packages/mysql/lib/mysql_p.js

```diff
@@ -66,16 +67,72 @@ function patchCreatePool(mysql) {
   };
 }
 
+function patchCreatePoolCluster(mysql) {
+  var baseFcn = '__createPoolCluster';
+  mysql[baseFcn] = mysql['createPoolCluster'];
+
+  mysql['createPoolCluster'] = function patchedCreatePoolCluster() {
+    var poolCluster = mysql[baseFcn].apply(poolCluster, arguments);
+    if (poolCluster.query instanceof Function) {
+      patchObject(poolCluster);
+    }
+    return poolCluster;
+  };
+}
+
+function patchOf(poolCluster) {
+  var baseFcn = '__of';
+  poolCluster[baseFcn] = poolCluster['of'];
+
+  poolCluster['of'] = function patchedOf() {
+    var args = arguments;
+
+    var resultPool = poolCluster[baseFcn].apply(poolCluster, args);
+    return patchObject(resultPool);
+  }
+}
+
+function patchGetConnection(pool) {
+  var baseFcn = '__getConnection';
+  pool[baseFcn] = pool['getConnection'];
+
+  pool['getConnection'] = function patchedGetConnection() {
+    var args = arguments;
+    var callback = args[args.length-1];
+
+    if (callback instanceof Function) {
+      args[args.length-1] = (err, connection) => {
+        if(connection) patchObject(connection);
+        return callback(err, connection);
+      }
+    }
+
+    var result = pool[baseFcn].apply(pool, args);
+    if (result instanceof Promise) return result.then(patchObject);
```

Comment:
Sorry I missed this before, but could you instead check if `result` has a method `then`? I've seen this check bite us in other libraries when a user has replaced the native `Promise` class with a different once, e.g. bluebird.